packages/react-testing/src/root.tsx

```diff
@@ -0,0 +1,249 @@
+import * as React from 'react';
+import {render, unmountComponentAtNode} from 'react-dom';
+import {act} from 'react-dom/test-utils';
+import {
+  Arguments,
+  Props as PropsForComponent,
+  MaybeFunctionReturnType as ReturnType,
+} from '@shopify/useful-types';
+
+import {TestWrapper} from './TestWrapper';
+import {Element, Predicate} from './element';
+import {Tag, Fiber, ReactInstance, FunctionKeys} from './types';
+import {withIgnoredReactLogs} from './errors';
+
+// eslint-disable-next-line typescript/no-var-requires
+const {findCurrentFiberUsingSlowPath} = require('react-reconciler/reflection');
+
+export const connected = new Set<Root<unknown>>();
+
+export class Root<Props> {
+  get props() {
+    return this.withRoot(root => root.props);
+  }
+
+  get isDOM() {
+    return this.withRoot(root => root.isDOM);
+  }
+
+  get instance() {
+    return this.withRoot(root => root.instance);
+  }
+
+  get children() {
+    return this.withRoot(root => root.children);
+  }
+
+  get descendants() {
+    return this.withRoot(root => root.descendants);
+  }
+
+  get domNodes() {
+    return this.withRoot(root => root.domNodes);
+  }
+
+  get domNode() {
+    return this.withRoot(root => root.domNode);
+  }
+
+  private wrapper: TestWrapper<Props> | null = null;
+  private element = document.createElement('div');
+  private root: Element<Props> | null = null;
+
+  private get mounted() {
+    return this.wrapper != null;
+  }
+
+  constructor(private tree: React.ReactElement<Props>) {
+    this.mount();
+  }
+
+  perform<T>(action: () => T, {update = true} = {}): T {
```

Comment:
Any reason not to call this `act` for symmetry with react's API?

Reply:
I don't like the act name :P but you're right, better to match them